File: src/types/rag.ts
export interface SourceCitation {
  source_id?: string | null;
  source_name: string;
  source_type: 'website' | 'pdf' | 'image' | 'enrichment' | 'product_metadata' | string;
  page_number?: number | null;
  url?: string | null;
  evidence_text: string;
  similarity_score?: number;
}

export interface ChunkMetadata {
  chunk_id: string;
  product_id: string;
  document_id?: string | null;
  source_id?: string | null;
  source_type: string;
  source_name: string;
  page_number?: number | null;
  url?: string | null;
}

export interface VectorChunk {
  chunk_id: string;
  text: string;
  metadata: ChunkMetadata;
  similarity_score?: number;
}

export interface IndexProductResponse {
  product_id: string;
  status: string;
  indexed_chunks_count: number;
  collection_name: string;
  message: string;
}

export interface VectorSearchResponse {
  product_id: string;
  query: string;
  results_count: number;
  chunks: VectorChunk[];
}

export interface VerifyAttributeResponse {
  product_id: string;
  attribute_name: string;
  value: string;
  confidence: number;
  verified: boolean;
  evidence_text: string;
  supporting_sources: SourceCitation[];
}

export interface ProductQAResponse {
  product_id: string;
  question: string;
  answer: string;
  found_evidence: boolean;
  citations: SourceCitation[];
}

File: backend/app/schemas/rag.py
from pydantic import BaseModel, Field
from typing import List, Optional, Dict, Any

class SourceCitation(BaseModel):
    source_id: Optional[str] = None
    source_name: str
    source_type: str  # 'website', 'pdf', 'image', 'enrichment'
    page_number: Optional[int] = None
    url: Optional[str] = None
    evidence_text: str
    similarity_score: Optional[float] = 0.0

class ChunkMetadata(BaseModel):
    chunk_id: str
    product_id: str
    document_id: Optional[str] = None
    source_id: Optional[str] = None
    source_type: str
    source_name: str
    page_number: Optional[int] = None
    url: Optional[str] = None

class VectorChunk(BaseModel):
    chunk_id: str
    text: str
    metadata: ChunkMetadata
    similarity_score: Optional[float] = 0.0

class IndexProductResponse(BaseModel):
    product_id: str
    status: str
    indexed_chunks_count: int
    collection_name: str
    message: str

class VectorSearchRequest(BaseModel):
    query: str
    top_k: int = Field(default=5, ge=1, le=20)
    min_score: float = Field(default=0.0, ge=0.0, le=1.0)

class VectorSearchResponse(BaseModel):
    product_id: str
    query: str
    results_count: int
    chunks: List[VectorChunk]

class VerifyAttributeRequest(BaseModel):
    attribute_name: str
    current_value: Optional[str] = None

class VerifyAttributeResponse(BaseModel):
    product_id: str
    attribute_name: str
    value: str
    confidence: float
    verified: bool
    evidence_text: str
    supporting_sources: List[SourceCitation]

class ProductQARequest(BaseModel):
    question: str
    top_k: int = 5

class ProductQAResponse(BaseModel):
    product_id: str
    question: str
    answer: str
    found_evidence: bool
    citations: List[SourceCitation]
